Sources/SpeziBluetooth/Utils/AsyncSempahore.swift

```diff
@@ -0,0 +1,145 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+
+
+final class AsyncSemaphore {
```

Comment:
It might make sense to move some of these Util types to Spezi Foundation in the future?

Reply:
Yep, we could move that to SpeziFoundation. I can create an issue for that in SpeziFoundation 👍

Reply:
This is now tracked in https://github.com/StanfordSpezi/SpeziFoundation/issues/5